lib/tasks/troubleshoot.rake

```diff
@@ -39,13 +39,26 @@ namespace :troubleshoot do
     DeliveryRequestWorker.perform_async_in_queue(email.id, queue: :delivery_immediate)
   end
 
-  desc "Re-send failed emails by email ids"
-  task resend_failed_emails: [:environment] do |_, args|
-    failed_email_ids = Email.where(id: args.to_a, status: "failed").pluck(:id)
+  namespace :resend_failed_emails do
+    desc "Re-send failed emails by email ids"
+    task by_id: [:environment] do |_, args|
+      resend_failed_emails(Email.where(id: args.to_a))
+    end
 
-    failed_email_ids.each do |email_id|
-      puts "Resending email: #{email_id}"
-      DeliveryRequestWorker.perform_async_in_queue(email_id, queue: :delivery_immediate)
+    desc "Re-send failed emails by date range"
+    task :by_date, %i[from to] => [:environment] do |_, args|
+      from = Time.iso8601(args.fetch(:from))
+      to = Time.iso8601(args.fetch(:to))
+      resend_failed_emails(Email.where(created_at: from..to))
     end
   end
 end
+
+def resend_failed_emails(scope)
+  ids = scope.where(status: :failed).pluck(:id)
+  puts "Resending #{ids.length} emails"
+
+  ids.each do |id|
+    DeliveryRequestWorker.perform_async_in_queue(id, queue: :delivery_immediate)
```

Comment:
Do we also potentially want to add an option to specify the `delivery_immediate_high` queue too?

Reply:
Thinking about it some more, I think it actually makes sense for these tasks to default to the high queue, as they're designed to be used to resolve an incident. I've made that change in 52e9d1b. I haven't added Rake tasks to add them to the normal priority queue as we can always add them later if we discover that we need them, but also it helps to avoid people having to decide which one to use in the context of an incident.

Reply:
Sounds reasonable to me!